### PR TITLE
ROX-21943: Add discovered-clusters route in UI

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

Adapt from
https://github.com/stackrox/stackrox/tree/master/ui/apps/platform/src/Containers/Administration/Events

Follow procedure
https://github.com/stackrox/stackrox/tree/master/ui/apps/platform#add-a-route

1. Edit ui/apps/platform/src/routePaths.ts file.
    * Add a path **without** params for link from sidebar navigation ~and, if needed, path **with** params for the `Route` element~.
        Because table row displays all information about a discovered cluster, no separate page for MVP.
    * Add a string to `RouteKey` type.
    * Add a property to `routeRequirementsMap` object.
        Specify `ROX_CLOUD_SOURCES` feature flag during development of a new route.
        Specify `Administration` minimum resource requirements. Because **Discovered clusters** link is on clusters page, `Cluster` is an implicit requirement.
2. Edit ui/apps/platform/src/Containers/MainPage/Body.tsx file.
    * Import the path for the `Route` element.
    * Add a property to `routeComponentMap` object.
3. ~Edit ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationSidebar.tsx file, if the route has a link.~
4. Add a folder and root component file (see step 2).
    * Add src/Containers/Clusters/DiscoveredClusters folder and DiscoveredClustersPage.tsx file.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js 461 = 3949941 - 3949480
        total 10445 = 10432581 - 10422136
    * `ls -al build/static/js/*.js | wc`
        files 1 = 92 - 91
3. `yarn start` in ui

#### without feature flag

1. Visit /main/clusters/discovered-clusters
    Instead of 404 page, see error on clusters page because it interprets discovered-clusters as an id by mistake :(

#### with feature flag

Before you `yarn deploy` in ui folder:

```sh
export ROX_CLOUD_SOURCES=true
```

1. Visit /main/clusters/discovered-clusters
    See **Discovered clusters** page.

2. Click **Clusters** link in breadcrumbs

### CODE REVIEW

File: ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    const [count, setCount] = useState(0);
+    const [errorMessage, setErrorMessage] = useState('');
+    const [clusters, setClusters] = useState<DiscoveredCluster[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+        setIsLoading(true);
+
+        // const listArg = getListDiscoveredClustersArg({ page, perPage, searchFilter, sortOption });
+        // const { filter } = listArg;
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx:48]: Is this comment a TODO, for functionality to be added later?

File: ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        Promise.all([countDiscoveredClusters(/* filter */), listDiscoveredClusters(/* listArg */)])
+            .then(([countArg, clustersArg]) => {
+                setClusters(clustersArg);
+                setCount(countArg);
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx:53]: nit: I initial read the `...Arg` part of these var names as "Argument" which confused me. Maybe "Response" would be better suffix.

```suggestion
            .then(([countResponse, clustersResponse]) => {
                setClusters(clustersResponse);
                setCount(countResponse);
```

**Review by vjwilson — APPROVED** [review]
Good iterative approach. Just minor questions

File: ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx:48]: Yes, commented out pending details about discovered clusters service.

File: ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx (hunk 2 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx]: Thank you for comments from reader viewpoint. I renamed with **FromResponse** suffix.

**Pull request merged**